lib/components/src/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,200 @@
+
+type AccordionMap = Record<string, { index: number; id: string }>;
+
+type StateChange = {
+  id: string;
+  index: number;
+};
+
+export type AccordionProps = {
+  allowMultipleOpen?: boolean;
+  defaultOpen?: number | number[];
+  bordered?: boolean;
+  rounded?: boolean;
+  lined?: boolean;
+  onOpen?: (change: StateChange) => void;
+  onClose?: (change: StateChange) => void;
+} & React.HTMLAttributes<HTMLUListElement>;
+
+/**
+ *
+ * Controlling wrapper component that will control the open/close functionality to
+ * child ```<AccordionItem />```'s and style the other support components
+ * ```<AccordionHeader />``` and ```<AccordionBody />```
+ *
+ * Although each component can work independently and be controlled from the outside
+ * via an **open** prop, it is not recommended to use ```<AccordionItem />```,
+ *  ```<AccordionHeader />``` and ```<AccordionBody />``` without this controller
+ *
+ * If no ```<AccordionBody />``` is inside an ```<AccordionItem />``` the open/close
+ * functionality will be removed by default, as well as the icon and pointer cursor.
+ */
+export const Accordion = ({
+  allowMultipleOpen,
+  children,
+  defaultOpen,
+  rounded = false,
+  bordered = false,
+  lined = false,
+  onOpen = () => {},
+  onClose = () => {},
+  ...rest
+}: AccordionProps) => {
+  const [open, setOpen] = useState<OpenMap>({});
+  const itemMap = useRef<AccordionMap>({});
+  const itemMapIndex = useRef(0);
+  const openMap = useRef<OpenMap>({});
+
+  const addToMap: AddToMapFn = useCallback(
+    (id) => {
+      const index = itemMapIndex.current + 1;
+      itemMap.current = { ...itemMap.current, [id]: { id, index } };
+      itemMapIndex.current = index;
+
+      if (typeof defaultOpen === 'number') {
+        openMap.current = { ...openMap.current, [id]: defaultOpen === index };
+      } else if (Array.isArray(defaultOpen) && allowMultipleOpen) {
+        const isExpanded = defaultOpen.includes(index);
+        openMap.current = { ...openMap.current, [id]: isExpanded };
+      } else {
+        openMap.current = { ...openMap.current, [id]: false };
+      }
+    },
+    [itemMap, openMap]
+  );
+
+  const onItemClose = useCallback(
+    (id: string) => {
+      setOpen({ ...open, [id]: false });
+      onClose({ id, index: itemMap.current[id].index });
+    },
+    [open, setOpen, onOpen, itemMap]
+  );
+
+  const onItemExpand = useCallback(
+    (id: string) => {
+      const newOpen = { ...open };
+      let oldOpen: StateChange | undefined;
+
+      if (allowMultipleOpen) {
+        newOpen[id] = true;
+      } else {
+        Object.keys(open).forEach((key) => {
+          if (open[key]) {
+            oldOpen = { id: key, index: itemMap.current[key].index };
+          }
+
+          newOpen[key] = id === key;
+        });
+      }
+
+      if (oldOpen) {
+        onClose(oldOpen);
+      }
+
+      setOpen({ ...open, ...newOpen });
+      onOpen({ id, index: itemMap.current[id].index });
+    },
+    [open, setOpen, itemMap]
```

Comment:
You use onOpen here, but it's not added to the useCallback dependency?

I suggest not having a default value for onOpen, but instead do something like:

```
onOpen && onOpen({ id, index: itemMap.current[id].index });
```